src/__tests__/erc721/erc721.test.ts

```diff
@@ -0,0 +1,267 @@
+import core from '../../core';
+import erc165 from '../../erc165';
+import erc20 from '../../erc20';
+import erc721 from '../../erc721';
+import { testGraphql } from '../utils';
+
+const { execQuery } = testGraphql({ optsOverride: { plugins: [core, erc20, erc165, erc721] } });
+
+test('erc721: nftToken balanceOf query #1', async () => {
+  const query = `
+  {
+    nftToken(address:"0x06012c8cf97BEaD5deAe237070F9587f8E7A266d") {
+      balanceOf(owner: "0xD418c5d0c4a3D87a6c555B7aA41f13EF87485Ec6")
+    }
+  }
+  `;
+  const result = await execQuery(query);
+  expect(result.errors).toBeUndefined();
+  expect(result.data).not.toBeUndefined();
+
+  expect(result.data).toEqual({
+    nftToken: {
+      balanceOf: 0,
+    },
+  });
+});
+
+test('erc721: nftToken balanceOf query #2', async () => {
+  const query = `
+  {
+    nftToken(address:"0x06012c8cf97BEaD5deAe237070F9587f8E7A266d") {
+      balanceOf(owner: "0x595a6aA36Ab9fFB4b5940D4E189d6F2AB3958aeb")
+    }
+  }
+  `;
+  const result = await execQuery(query);
+  expect(result.errors).toBeUndefined();
+  expect(result.data).not.toBeUndefined();
+
+  expect(result.data).toEqual({
+    nftToken: {
+      balanceOf: 72,
+    },
+  });
+});
+
+test('erc721: nftToken ownerOf query', async () => {
+  const query = `
+  {
+    nftToken(address:"0x06012c8cf97BEaD5deAe237070F9587f8E7A266d") {
+      ownerOf(tokenId: 384978)
+    }
+  }
+  `;
+  const result = await execQuery(query);
+  expect(result.errors).toBeUndefined();
+  expect(result.data).not.toBeUndefined();
+
+  expect(result.data).toEqual({
+    nftToken: {
+      ownerOf: '0x595a6aA36Ab9fFB4b5940D4E189d6F2AB3958aeb',
+    },
+  });
+});
+
+test('erc721: nftToken getApproved query', async () => {
+  const query = `
+  {
+    nftToken(address:"0x6EbeAf8e8E946F0716E6533A6f2cefc83f60e8Ab") {
+      getApproved(tokenId: 33525)
```

Comment:
We shy away from mapping 1:1 functions to fields. We prefer to represent the data model. In this case, it could look like this:

```
{
  nftContract {
    token(id: 12345) {
      approved    # this needs to return an array of Address
      owner         # this needs to return an Address
      ...
    }
  }
}
```

Reply:
Hey, well I just defined defined tried to to resemble the most ERC721.  what you suggest is good for the function `ownerOf` and `getApproved`, but the functions like `balanceOf` and `isApprovedForAll` doesn't receive the `tokenId`